java/code/src/com/redhat/rhn/domain/user/UserFactory.java

```diff
@@ -548,16 +548,20 @@ public int compare(Object o1, Object o2) {
                         }
 
                         if (offSet1 <= -18000000 && offSet1 >= -36000000 &&
-                                offSet2 <= -18000000 && offSet2 >= -36000000) {
+                                offSet2 <= -18000000 && offSet2 >= -36000000 &&
+                                t1.getOlsonName().startsWith("America") &&
```

Comment:
These heuristics end up switching Hawaii and Midway. Canada(Regina) is already in the middle of the 'United States' set. I'm not sure this little piece of US-centrism is worth the fragility. Maybe modify the comparator so that it puts GMT first and then just lists everything east-to-west from there?